mmv1/third_party/terraform/resources/resource_container_node_pool.go.erb

```diff
@@ -378,9 +378,25 @@ func resourceContainerNodePoolCreate(d *schema.ResourceData, meta interface{}) e
 	if err != nil {
 		return err
 	}
-
+<% if version == "ga" -%>	
 	mutexKV.Lock(nodePoolInfo.lockKey())
 	defer mutexKV.Unlock(nodePoolInfo.lockKey())
+<% else -%>
+	clusterHash, err := getClusterHash(nodePoolInfo.parent(), nodePoolInfo.project, userAgent, config)
+	if err != nil {
+			return err
+	}
+
+	// Acquire read-lock on cluster.
```

Comment:
Just to make sure I understand this right- we're taking a Read lock on the cluster & this won't block other node pools from moving forward (and they'll be able to take additional read locks), but *will* block cluster-level operations because they'll take a write lock, which blocks until all read locks are released.

Reply:
Yep, that's correct.